addons/website/models/res_users.py

```diff
@@ -62,10 +62,19 @@ def authenticate(cls, db, login, password, user_agent_env):
             with cls.pool.cursor() as cr:
                 env = api.Environment(cr, uid, {})
                 visitor_sudo = env['website.visitor']._get_visitor_from_request()
+                partner_id = env.user.partner_id.id
                 if visitor_sudo:
-                    vals = {
-                        'user_partner_id': env.user.partner_id.id,
-                        'name': env.user.partner_id.name
-                    }
-                    visitor_sudo.write(vals)
+                    visitor_partner = env['website.visitor'].sudo().search([('user_partner_id', '=', partner_id)])
+                    if visitor_partner and visitor_partner.id != visitor_sudo.id:
```

Comment:
if visitor_partner is null that means there is no previous visitor for this partner. I don't want to reassign the visitor's track/last_connections in that case since there is no one to reassign them too.